datadog_checks_dev/datadog_checks/dev/tooling/commands/meta/snmp/build_traps_db.py

```diff
@@ -0,0 +1,294 @@
+
+        compiled_mibs_sources = os.path.abspath(compiled_mibs_sources)
+        if not os.path.isdir(compiled_mibs_sources):
+            abort("Output directory {} does not exist".format(compiled_mibs_sources))
+        echo_info("Writing intermediate compiled MIBs to {}".format(compiled_mibs_sources))
+
+        mibs_sources_dir = os.path.join(compiled_mibs_sources, 'mibs_sources')
+        if not os.path.isdir(mibs_sources_dir):
+            os.mkdir(mibs_sources_dir)
+
+        if output_dir and output_file:
+            abort("Do not set both --output-dir and --output-file at the same time.")
+        elif not output_file and not output_dir:
+            abort("Need to set one of --output-dir or --output-file")
+
+        mib_sources = (
+            sorted(set([os.path.abspath(os.path.dirname(x)) for x in mib_files if os.path.sep in x])) + mib_sources
+        )
+
+        mib_files = [os.path.basename(os.path.splitext(x)[0]) for x in mib_files if os.path.isfile(x)]
+        searchers = [AnyFileSearcher(compiled_mibs_sources).setOptions(exts=['.json'])]
+        code_generator = JsonCodeGen()
+        file_writer = FileWriter(compiled_mibs_sources).setOptions(suffix='.json')
+        mib_compiler = MibCompiler(SmiV1CompatParser(tempdir=''), code_generator, file_writer)
+        mib_compiler.addSources(*getReadersFromUrls(*mib_sources, **dict(fuzzyMatching=True)))
+        mib_compiler.addSearchers(*searchers)
+
+        try:
+            compiled_mibs, compiled_dependencies_mibs = compile_and_report_status(mib_files, mib_compiler)
+        except error.PySmiError:
+            abort("Unable to compile")
+
+        # Move all the parent MIBs that had to be compiled but were not requested in the command to a subfolder.
+        for mib_file_name in compiled_dependencies_mibs:
+            os.replace(
+                os.path.join(compiled_mibs_sources, mib_file_name + '.json'),
+                os.path.join(mibs_sources_dir, mib_file_name + '.json'),
+            )
+
+        # Only generate trap_db with `mib_files` unless explicitly asked. Used to ignore other files that may be
+        # present "compiled_mibs_sources"
+        compiled_mibs = [os.path.join(compiled_mibs_sources, x + '.json') for x in compiled_mibs]
+
+        # Generate the trap database based on the compiled MIBs.
+        trap_db_per_mib = generate_trap_db(compiled_mibs, mibs_sources_dir)
+
+        if output_file:
+            # Compact representation, only one file
+            write_compact_trap_db(trap_db_per_mib, output_file, use_json=use_json)
+            echo_success("Wrote trap data to {}".format(os.path.abspath(output_file)))
+        else:
+            # Expanded representation, one file per MIB.
+            write_trap_db_per_mib(trap_db_per_mib, output_dir, use_json=use_json)
+            echo_success("Wrote trap data to {}".format(os.path.abspath(output_dir)))
+
+
+def compile_and_report_status(mib_files, mib_compiler):
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities. -->
**💭 thought**
In a follow up PR, this could be moved to some common utils file to be used by both profile generator and traps builder